packages/accounts/src/msca/plugins/multisig/signer.ts

```diff
@@ -0,0 +1,106 @@
+import type {
+  Address,
+  BundlerClient,
+  SmartAccountSigner,
+} from "@alchemy/aa-core";
+import {
+  hashMessage,
+  hashTypedData,
+  type Hash,
+  type Hex,
+  type SignableMessage,
+  type Transport,
+  type TypedData,
+  type TypedDataDefinition,
+} from "viem";
+import { MultisigPlugin, MultisigPluginAbi } from "./plugin.js";
+
+type MultisigMessageSignerParams<
+  TTransport extends Transport,
+  TSigner extends SmartAccountSigner
+> = {
+  client: BundlerClient<TTransport>;
+  accountAddress: Address;
+  signer: () => TSigner;
+  threshold: bigint;
+  pluginAddress?: Address;
+};
+
+export const multisigMessageSigner = <
+  TTransport extends Transport,
+  TSigner extends SmartAccountSigner
+>({
+  client,
+  accountAddress,
+  signer,
+  threshold,
+  pluginAddress = MultisigPlugin.meta.addresses[client.chain.id],
+}: MultisigMessageSignerParams<TTransport, TSigner>) => {
+  const signWith712Wrapper = async (msg: Hash): Promise<`0x${string}`> => {
+    const [, name, version, chainId, verifyingContract, salt] =
+      await client.readContract({
+        abi: MultisigPluginAbi,
+        address: pluginAddress,
+        functionName: "eip712Domain",
+        account: accountAddress,
+      });
+
+    return signer().signTypedData({
+      domain: {
+        chainId: Number(chainId),
+        name,
+        salt,
+        verifyingContract,
+        version,
+      },
+      types: {
+        AlchemyMultisigMessage: [{ name: "message", type: "bytes" }],
+      },
+      message: {
+        message: msg,
+      },
+      primaryType: "AlchemyMultisigMessage",
+    });
+  };
+
+  return {
+    getDummySignature: async (): Promise<`0x${string}`> => {
+      const [, thresholdRead] = await client.readContract({
+        abi: MultisigPluginAbi,
+        address: pluginAddress,
+        functionName: "ownershipInfoOf",
+        args: [accountAddress],
+      });
+
+      const actualThreshold = thresholdRead === 0n ? threshold : thresholdRead;
+
+      // todo: revert on account client creation if threshold is zero
```

Comment:
should we still handle this?

Reply:
Not sure. It will result in a revert on `getAddress` from the factory, so we can't calculate a counterfactual for it. Is that enough to handle the case, or should we add additional checking here?

Reply:
yea that sounds fine, if you want to manually check this that's fine too and then throw one of the class based errors. if we put together a doc then that error could link out to the doc

Reply:
removed the comment